native/vendor/google/play/TResult.h

```diff
@@ -59,5 +59,110 @@ class RecallAccess : public RefCounted {
     std::string _sessionId;
 };
 
+class Achievement {
+public:
+    
+    int getCurrentSteps() const {
+        return _currentSteps;
+    }
+    int getState() const {
+        return _state;
+    }
+    int getTotalSteps() const {
+        return _totalSteps;
+    }
+    int getType() const {
+        return _type;
+    }
+    long getLastUpdatedTimestamp() const {
+        return _lastUpdatedTimestamp;
+    }
+    long getXpValue() const {
+        return _xpValue;
+    }
+    const std::string& getAchievementId() const {
+        return _achievementId;
+    }
+    const std::string& getDescription() const {
+        return _description;
+    }
+    const std::string& getFormattedCurrentSteps() const {
+        return _formattedCurrentSteps;
+    }
+    const std::string& getFormattedTotalSteps() const {
+        return _formattedTotalSteps;
+    }
+    const std::string& getName() const {
+        return _name;
+    }
+    const std::string& getRevealedImageUrl() const {
+        return _revealedImageUrl;
+    }
+    const std::string& getUnlockedImageUrl() const {
+        return _unlockedImageUrl;
+    }
+private:
+    friend class PlayTask;
+    static const char TYPE_INCREMENTAL{1};
+    int _currentSteps{0};
+    int  _state{0};
+    int _totalSteps{0};
+    int _type{0};
+    long _lastUpdatedTimestamp{0};
+    long _xpValue{0};
+    std::string _achievementId;
+    std::string _description;
+    std::string _formattedCurrentSteps;
+    std::string _formattedTotalSteps;
+    std::string _name;
+    std::string _revealedImageUrl;
+    std::string _unlockedImageUrl;
+};
+
+class AchievementBuffer {
+public:
+    ~AchievementBuffer() {
+        for(auto* achievement : _achievements) {
+            delete achievement;
+        }
+        _achievements.clear();
+    }
+    size_t getCount() const {
+        return _achievements.size();
+    }
+    const Achievement* get(int i) {
+        if(i >= 0 && i < _achievements.size()) {
+            return _achievements[i];
+        }
+        return nullptr;
+    }
+    void close() {
+        release();
+    }
+    bool isCloseed() {
+        CC_LOG_WARNING("interface not implemented");
+        return false;
+    }
+    void release() {
+        CC_LOG_WARNING("interface not implemented");

```

Comment:
Any reason that we can't support `release` interface  since it's avaiable in java : https://developers.google.com/android/reference/com/google/android/gms/common/data/AbstractDataBuffer#release()